### PR TITLE
Readthedocs

### PR DESCRIPTION
**Pull request by bentheiii**

### CODE REVIEW

File: docs/clients.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    .. method:: set_defaults(**kwargs: str | contextvars.ContextVar[str])
+
+        Sets the default context feature values. The keys of *\*\*kwargs* should be context feature names. The values
+        can be either an :class:`str`, in which case default value for the context feature will be the string.
+        Alternatively, the default value may be a context variable, in which case the value will be fetched dynamically.
+
+    .. method:: ping()->None
+        :async:
+
+        Pings the Heksher service.
```
Comment (asafalon1) [inline docs/clients.rst:68]: and? what happens when the ping fails?

File: docs/clients.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    .. method:: set_defaults(**kwargs: str | contextvars.ContextVar[str])
+
+        Sets the default context feature values. The keys of *\*\*kwargs* should be context feature names. The values
+        can be either an :class:`str`, in which case default value for the context feature will be the string.
+        Alternatively, the default value may be a context variable, in which case the value will be fetched dynamically.
+
+    .. method:: ping()->None
+
+        Pings the Heksher service.
```
Comment (asafalon1) [inline docs/clients.rst:153]: same as previous comment

File: docs/stub_clients.rst [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    A rule used to patch a setting's value depending on context features.
+
+    This class is a dataclass.
+
+    :param match_conditions: A mapping of context feature names to their exact-match conditions.
+
+        .. note::
+
+            In sub clients, when multiple rules are given, they must have the exact same keys, in exactly the same
```
Comment (asafalon1) [inline docs/stub_clients.rst:115]: you mean stub clients?

**Review by asafalon1 — DISMISSED** [review]

File: docs/cookbook.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+To fix this, we'll use the "alias" parameter when declaring the setting.
+
+.. code-block:: python
+
+    bar = Setting('bar', int, ..., alias='foo')
+
+This will inform the heksher service that the setting known as ``foo`` should be renamed to bar, along with all the
+rules associated with it. As an added bonus, any app that uses the older name will not break, as they will be able to
+access ``bar`` via its alias (and old name) ``foo``.
```
Comment (MarvaZy) [inline docs/cookbook.rst:27]: what about multiple aliases?

File: docs/cookbook.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+Offline Settings
+====================
+
+Sometimes we may want to have the Heksher client to be able retrieve the rules from an online heksher service, but we
```
Comment (MarvaZy) [inline docs/cookbook.rst:32]: we may want to have the Heksher client to be able _**to**_ retrieve...

File: docs/cookbook.rst (hunk 4 above)
Comment (bentheiii) [reply docs/cookbook.rst:27]: not supported

Comment (MarvaZy) [reply docs/cookbook.rst:27]: I think you should write it, as well as if you can change a third time (by "aliasing" the new name)?

**Review by MarvaZy — APPROVED** [review]

**Pull request merged**